packages/backend/src/repositories/product.ts

```diff
@@ -65,6 +69,27 @@ export class ProductRepository {
     ]);
   }
 
+  public search({ q }: ProductSearchQuery): Promise<ProductSearchResponse[]> {
+    return this._dbClient.$queryRaw`
+      SELECT
+        "Product"."id",
+        "Product"."title"
+      FROM
+        "Product"
+      WHERE 
+        (SIMILARITY("Product"."title", ${q}) > 0.2
+      OR 
+        "Product"."title" ILIKE ${q + '%'}
+      OR 
+        "Product"."title" ILIKE ${'%' + q + '%'})
```

Comment:
We will find similarity only by title, not by description?

Reply:
at this moment i think its enough, description can contain random data which doesnt match the search query

Reply:
ok